In [1]:
--1. Write a query to get the sum of impressions by day.
SELECT  CONVERT(DATE, [date]) AS Date
    , SUM([Impressions]) AS Daily_Impressions
FROM dbo.Marketing_Performance
GROUP BY [date]
ORDER BY [date]
-- I have [ ] because I dragged the column name straight from the directory, which is faster and more accurate than typing.


(17 rows affected)

Total execution time: 00:00:00.046

Date Daily_Impressions 2023-07-24 1423 2023-07-26 1547 2023-07-27 2295 2023-07-28 8142 2023-07-29 3511 2023-07-30 3001 2023-08-02 1532 2023-08-03 3189 2023-08-04 3254 2023-08-05 1088 2023-08-10 2128 2023-08-11 2278 2023-08-12 6588 2023-08-13 2170 2023-08-14 5773 2023-08-17 3292 2023-08-19 1248

In [3]:
-- 2. Write a query to get the top three revenue-generating states in order of best to worst.
SELECT TOP 3 [state] as State
    , SUM([Revenue]) AS Total_Revenue
FROM dbo.Website_Revenue
GROUP BY [state]
ORDER BY Total_Revenue DESC


(3 rows affected)

Total execution time: 00:00:00.038

State Total_Revenue NY 46398 GA 39666 OH 37577

<span style="color: #008000;">-- How much revenue did the third best state generate?</span>

<span style="color: #008000;">-- The third revenue-generating state is Ohio, and it generated $37577.</span>

In [5]:
--3. Write a query that shows total cost, impressions, clicks, and revenue of each campaign.
-- Make sure to include the campaign name in the output.

SELECT c.[name] AS Campaign_Name --Renaming all these columns
    , m.[campaign_id] AS Campain_ID
    , ROUND(SUM(m.[cost]),2) AS Total_Cost --Only want 2 decimal places
    , SUM(m.[impressions]) AS Total_Impressions --Finding the sum value for all these columns that need to be included.
    , SUM(m.[clicks]) AS Total_Clicks
    , SUM(w.[revenue]) AS Total_Revenue
FROM dbo.marketing_performance m --Renaming the csv files bc their names are too long.
INNER JOIN dbo.campaign_info c
ON m.[campaign_id]=c.[id]  -- Joining these 2 files first
INNER JOIN dbo.website_revenue w
ON m.[campaign_id]=w.[campaign_id] -- Then joining the joint file with the third file
GROUP BY c.[name], m.[campaign_id]
ORDER BY c.[name]

(5 rows affected)

Total execution time: 00:00:00.040

Campaign_Name Campain_ID Total_Cost Total_Impressions Total_Clicks Total_Revenue Campaign1 64441550 4170.51 42810 38724 151792 Campaign2 55304737 4075.5 40938 29652 155308 Campaign3 89363211 15809.04 158280 116048 551672 Campaign4 86363015 3970.14 47508 33318 163396 Campaign5 99058240 4077.15 25641 33663 136404

In [7]:
-- 4. Write a query to get the number of conversions of Campaign5 by state.
-- Which state generated the most conversions for this campaign?
SELECT c.[name] AS Campaign
    , m.[geo] AS State
    , SUM(m.[conversions]) AS Total_Conversions --Summing up number of conversions
FROM dbo.campaign_info c
INNER JOIN dbo.marketing_performance m
ON c.[id] = m.[campaign_id]
WHERE c.[name] = 'Campaign5' --filtering just for conversions in Campaign 5
GROUP BY m.[geo]
        ,c.[name]

-- Georgia generated the most conversoins for Campaign 5


(2 rows affected)

Total execution time: 00:00:00.045

Campaign State Total_Conversions Campaign5 United States-GA 672 Campaign5 United States-OH 442

In [1]:
-- 5. In your opinion, which campaign was the most efficient, and why?
SELECT c.[name] AS Campaign
    , SUM(m.[conversions]) AS Total_Conversions --Summing up number of conversions
FROM dbo.campaign_info c
INNER JOIN dbo.marketing_performance m
ON c.[id] = m.[campaign_id]
GROUP BY c.[name]
ORDER BY SUM(m.[conversions]) DESC

-- Campaign 3 seems to be the most effective overall because it has the highest sum of conversions across states.


(5 rows affected)

Total execution time: 00:00:00.105

Campaign Total_Conversions Campaign3 4451 Campaign1 3018 Campaign4 1551 Campaign2 1516 Campaign5 1114

In [12]:
--**Bonus Question**
--6. Write a query that showcases the best day of the week (e.g., Sunday, Monday, Tuesday, etc.) to run ads.

SELECT DATEPART(WEEKDAY, [date]) 'AS Day of the Week'  -- Drawing out the days of the week, according to the date
    , AVG([cost]/[conversions]) AS 'Dollars Per Conversion' -- Finding out how much each conversion costs.
FROM dbo.marketing_performance
GROUP BY DATEPART(WEEKDAY, [date])
ORDER BY AVG([cost]/[conversions]) ASC

-- Wednesday, the 4th day of the week, is the best day to run ads as it has the lowest cost per conversion.
-- I chose to compute the cost per conversion instead of cost per impression because it's more substantial and leads to real results

(6 rows affected)

Total execution time: 00:00:00.038

AS Day of the Week Dollars Per Conversion 4 0.3634327970939284 6 0.3811430773961086 5 0.4774955182406679 7 0.5062864191087108 2 0.5515364632092459 1 0.6358254244305318